### PR TITLE
GRAPHICS: Indeo 3 performance and memory improvements

### PR DESCRIPTION
**Pull request by ccawley2011**
This improves performance a bit on the DS, although currently the YUV to RGB code is the main bottleneck.

### CODE REVIEW

File: image/codecs/indeo3.cpp [hunk 1]
```diff
@@ -382,6 +382,28 @@ typedef struct {
 	}                     \
 	lp2 = 4;
 
+#define COPY_4_PIXELS(dst, src, count, pitch) \
+	if (IS_ALIGNED(src, 4)) { \
+		for (i = 0, j = 0; i < count; i++, j += pitch) \
+			dst[j] = src[j]; \
+	} else { \
+		for (i = 0, j = 0; i < count; i++, j += pitch) \
+			dst[j] = READ_UINT32(src+j); \
```
Comment (bluegr) [inline image/codecs/indeo3.cpp:391]:
```suggestion
			dst[j] = READ_UINT32(src + j); \
```

File: image/codecs/indeo3.cpp [hunk 2]
```diff
@@ -382,6 +382,28 @@ typedef struct {
...
+	}
+
+#define COPY_8_PIXELS(dst, src, count, pitch) \
+	if (IS_ALIGNED(src, 4)) { \
+		for (i = 0, j = 0; i < count; i++, j += pitch) {\
+			dst[j] = src[j]; \
+			dst[j+1] = src[j+1]; \
```
Comment (bluegr) [inline image/codecs/indeo3.cpp:398]:
```suggestion
			dst[j+1] = src[j + 1]; \
```

File: image/codecs/indeo3.cpp [hunk 3]
```diff
@@ -382,6 +382,28 @@ typedef struct {
...
+		} \
+	} else { \
+		for (i = 0, j = 0; i < count; i++, j += pitch) { \
+			dst[j] = READ_UINT32(src+j); \
+			dst[j+1] = READ_UINT32(src+j+1); \
```
Comment (bluegr) [inline image/codecs/indeo3.cpp:403]:
```suggestion
			dst[j+1] = READ_UINT32(src + j + 1); \
```

Comment (bluegr) [inline image/codecs/indeo3.cpp:403]:
```suggestion
			dst[j] = READ_UINT32(src + j); \
			dst[j+1] = READ_UINT32(src + j + 1); \
```

File: image/codecs/indeo3.cpp [hunk 4]
```diff
@@ -468,18 +491,19 @@ void Indeo3Decoder::decodeChunk(byte *cur, byte *ref, int width, int height,
 			}
 		}
 
-		cur_frm_pos = cur + width * strip->ypos + strip->xpos;
+		cur_frm_pos = (uint32 *)(cur + width * strip->ypos + strip->xpos);
```
Comment (bluegr) [inline image/codecs/indeo3.cpp:494]: Will that work correctly on BE systems?

Comment (ccawley2011) [reply image/codecs/indeo3.cpp:494]: It should be OK. In the existing code, the same cast would be made later on, and the `FROM_LE_32` is used if the values in the array are modified.

**Comment by sev-** [comment]
Looks like a good change. Would it be possible to address those formatting issues?

**Comment by bluegr** [comment]
I went ahead and committed the whitespace fixes

**Comment by sev-** [comment]
Thank you!

**Pull request merged**